### PR TITLE
`Test` Profile editing

### PR DESCRIPTION
**Pull request by hilalnigarbasan**

### CODE REVIEW

File: packages/playwright/src/tests/profile-editing.spec.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    await page.getByRole('menuitem', { name: 'Profile' }).click(); 
+    await page.getByRole('tab', { name: 'Security' }).click();
+    await page.getByPlaceholder('currentPassword').click();
+    await page.getByPlaceholder('currentPassword').fill('1234512345Ad');
+    await page.getByPlaceholder('Password', { exact: true }).click();
+    await page.getByPlaceholder('Password', { exact: true }).fill('1234567Ak');
+    await page.getByPlaceholder('passwordConfirmation').click();
+    await page.getByPlaceholder('passwordConfirmation').fill('1234567Ak');
+    await page.getByRole('button', { name: 'Change Password' }).click();
+    await page.evaluate(() => window.scrollTo(0, 0));
```
Comment (7alip) [inline packages/playwright/src/tests/profile-editing.spec.ts:26]: How would you know if the password was truly changed? You test if the user can log in with the updated password. You should usually use `expect` at the end of your test blocks.

```ts
// login
// update password
// login with updated password
// expect() => successful login, for example check if the profile button has username
```

File: packages/playwright/src/tests/profile-editing.spec.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+test.describe('Profile Editing Tests', () => {
+  test('User updates password', async ({ page }) => {
+    await page.goto(getVercelUrl('kunsthalte'))
+    await page.getByRole('button', { name: 'EN' }).click();
+    await page.getByRole('link', { name: 'Sign in' }).click();
+    await page.getByTestId('input-email').click();
+    await page.getByTestId('input-email').fill('Admin8');
+    await page.getByTestId('input-password').click();
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();
```
Comment (7alip) [inline packages/playwright/src/tests/profile-editing.spec.ts:14]: These steps are used in all tests. I would create a `Dashboard.ts` page object model. 
There you could have `login(username: string, password: string) {}`  function and call this function in your tests. See login.spec.ts and Login.ts files

File: packages/playwright/src/tests/profile-editing.spec.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    await page.getByPlaceholder('currentPassword').click();
+    await page.getByPlaceholder('currentPassword').fill('1234567Ak');
+    await page.getByPlaceholder('Password', { exact: true }).click();
+    await page.getByPlaceholder('Password', { exact: true }).fill('1');
+    await page.getByPlaceholder('passwordConfirmation').click();
+    await page.getByPlaceholder('passwordConfirmation').fill('1');
+    await page.getByRole('button', { name: 'Change Password' }).click();
+    // const errorMessageSelector = 'text=The provided current password is invalid';
+    // const isVisible = await page.isVisible(errorMessageSelector);
+    // expect(isVisible).toBeTruthy();
```
Comment (7alip) [inline packages/playwright/src/tests/profile-editing.spec.ts:53]: Why these lines are commented out? Are tests failing with this assertion?

File: packages/playwright/src/tests/profile-editing.spec.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    await page.getByRole('button', { name: 'Admin8' }).click();
+    await page.getByRole('menuitem', { name: 'Profile' }).click();
+    await page.getByRole('tab', { name: 'Security' }).click();
+    await page.getByPlaceholder('currentPassword').click();
+    await page.getByPlaceholder('currentPassword').fill('1234567Ak');
+    await page.getByPlaceholder('Password', { exact: true }).click();
+    await page.getByPlaceholder('Password', { exact: true }).fill('1234512345Ad');
+    await page.getByPlaceholder('passwordConfirmation').click();
+    await page.getByPlaceholder('passwordConfirmation').fill('1234512345Ad');
+    await page.getByRole('button', { name: 'Change Password' }).click();
```
Comment (7alip) [inline packages/playwright/src/tests/profile-editing.spec.ts:76]: There is no assertion (`expect`) in this test block.

File: packages/playwright/src/tests/profile-editing.spec.ts [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();
+    await page.evaluate(() => window.scrollTo(0, 0));
+    await page.getByRole('button', { name: 'Admin8' }).click();
+    await page.getByRole('menuitem', { name: 'Profile' }).click();
+    await page.getByRole('tab', { name: 'Socials' }).click();
+    await page.getByPlaceholder('linkedin').click();
+    await page.getByPlaceholder('linkedin').fill('https://www.linkedin.com/in/williamhgates/');
+    await page.evaluate(() => { window.scrollTo(0, document.body.scrollHeight); }); 
+    await page.getByRole('button', { name: 'Save' }).click();
```
Comment (7alip) [inline packages/playwright/src/tests/profile-editing.spec.ts:99]: There is no assertion (`expect`) in this test block.

File: packages/playwright/src/tests/profile-editing.spec.ts [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();
+    await page.evaluate(() => window.scrollTo(0, 0));
+    await page.getByRole('button', { name: 'Admin8' }).click(); 
+    await page.getByRole('menuitem', { name: 'Profile' }).click();
+    await page.getByRole('tab', { name: 'Socials' }).click(); 
+    await page.getByPlaceholder('linkedin').click(); 
+    await page.getByPlaceholder('linkedin').fill('https://www.linkedin.com/in/williamhgates');
+    await page.evaluate(() => { window.scrollTo(0, document.body.scrollHeight); });
+    await page.getByRole('button', { name: 'Save' }).click();
```
Comment (7alip) [inline packages/playwright/src/tests/profile-editing.spec.ts:142]: There is no assertion (`expect`) in this test block.

File: packages/playwright/src/tests/profile-editing.spec.ts [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+  test ('User can display new credentials/ security ', async ({ page }) => {
+    await page.goto(getVercelUrl('kunsthalte'))
+    await page.getByRole('button', { name: 'EN' }).click();
+    await page.getByRole('link', { name: 'Sign in' }).click();
+    await page.getByTestId('input-email').click();
+    await page.getByTestId('input-email').fill('Admin8');
+    await page.getByTestId('input-password').click();
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();
```
Comment (7alip) [inline packages/playwright/src/tests/profile-editing.spec.ts:156]: There is no assertion (`expect`) in this test block.

File: packages/playwright/src/tests/profile-editing.spec.ts [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+    await page.getByRole('button', { name: 'Admin8' }).click();
+    await page.getByRole('menuitem', { name: 'Profile' }).click();
+    await page.getByRole('tab', { name: 'Socials' }).click();
+    const page1Promise = page.waitForEvent('popup');
```
Comment (7alip) [inline packages/playwright/src/tests/profile-editing.spec.ts:174]: What exactly is this promise and popup event for?

If it's necessary, I would give a more descriptive name to this variable, like: `popupEventPromise`, `waitingPopupPromise` etc.

**Review by 7alip — CHANGES_REQUESTED** [review]
Thanks

**Comment by hilalnigarbasan** [comment]
//How would you know if the password was truly changed? You test if the user can log in with the updated password. You should usually use expect at the end of your test blocks.

The case that the user can log in with the updated password was requested in the second acceptance criteria under ‘after editing able to access new credentials’.
That's why I tested it under the ‘User can display new credentials/ security’ test header.

<https://1drv.ms/i/s!AqbmDqNXtL_NhCGfDl7uogqv8KBo>
[https://blz04pap003files.storage.live.com/y4m2fIqH91YDJb-cUC9jXQyQs2FOR8wqKPItDjZo28IqvAQkfONZX7TCctl-PbqaLuZl1XDYt6xHun_-5xRctncyr8qZbriX9gNZacJugcW_v-dcxKz2TMT50KMeHy0_nmyb7PN__tXSYKIrJHTkVWcP9bxGwx3vVBCoZGO8nooZIIcLvJu8zND1dpyZTOWgc4U?width=200&height=150&cropmode=center]
[https://res.cdn.office.net/assets/mail/file-icon/png/cloud_blue_16x16.png]Screenshot 2024-08-16 at 20.00.28.png



//Why these lines are commented out? Are tests failing with this assertion?
Yes, the test was failing with this assertion but I found a new assertion instead.


________________________________
Gönderen: Talip Altas ***@***.***>
Gönderildi: 14 Ağustos 2024 Çarşamba 22:23
Kime: freedomcombination/monorepo ***@***.***>
Bilgi: Hilal Nigar BAŞAN ***@***.***>; Author ***@***.***>
Konu: Re: [freedomcombination/monorepo] profile editing test done (PR #1579)


@7alip requested changes on this pull request.

Thanks

________________________________

In packages/playwright/src/tests/profile-editing.spec.ts<https://github.com/freedomcombination/monorepo/pull/1579#discussion_r1717435571>:

> +    await page.getByTestId('input-email').fill('Admin8');
+    await page.getByTestId('input-password').click();
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();
+    await page.evaluate(() => window.scrollTo(0, 0));
+    await page.getByRole('button', { name: 'Admin8' }).click();
+    await page.getByRole('menuitem', { name: 'Profile' }).click();
+    await page.getByRole('tab', { name: 'Security' }).click();
+    await page.getByPlaceholder('currentPassword').click();
+    await page.getByPlaceholder('currentPassword').fill('1234512345Ad');
+    await page.getByPlaceholder('Password', { exact: true }).click();
+    await page.getByPlaceholder('Password', { exact: true }).fill('1234567Ak');
+    await page.getByPlaceholder('passwordConfirmation').click();
+    await page.getByPlaceholder('passwordConfirmation').fill('1234567Ak');
+    await page.getByRole('button', { name: 'Change Password' }).click();
+    await page.evaluate(() => window.scrollTo(0, 0));


How would you know if the password was truly changed? You test if the user can log in with the updated password. You should usually use expect at the end of your test blocks.

// login
// update password
// login with updated password
// expect() => successful login, for example check if the profile button has username

________________________________

In packages/playwright/src/tests/profile-editing.spec.ts<https://github.com/freedomcombination/monorepo/pull/1579#discussion_r1717439147>:

> +    await page.goto(getVercelUrl('kunsthalte'))
+    await page.getByRole('button', { name: 'EN' }).click();
+    await page.getByRole('link', { name: 'Sign in' }).click();
+    await page.getByTestId('input-email').click();
+    await page.getByTestId('input-email').fill('Admin8');
+    await page.getByTestId('input-password').click();
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();


These steps are used in all tests. I would create a Dashboard.ts page object model.
There you could have login(username: string, password: string) {} function and call this function in your tests. See login.spec.ts and Login.ts files

________________________________

In packages/playwright/src/tests/profile-editing.spec.ts<https://github.com/freedomcombination/monorepo/pull/1579#discussion_r1717440262>:

> +    // const errorMessageSelector = 'text=The provided current password is invalid';
+    // const isVisible = await page.isVisible(errorMessageSelector);
+    // expect(isVisible).toBeTruthy();


Why these lines are commented out? Are tests failing with this assertion?

________________________________

In packages/playwright/src/tests/profile-editing.spec.ts<https://github.com/freedomcombination/monorepo/pull/1579#discussion_r1717440892>:

> +    await page.getByRole('button', { name: 'EN' }).click();
+    await page.getByRole('link', { name: 'Sign in' }).click();
+    await page.getByTestId('input-email').click();
+    await page.getByTestId('input-email').fill('Admin8');
+    await page.getByTestId('input-password').fill('1234567Ak');
+    await page.getByTestId('button-submit-login').click();
+    await page.getByRole('button', { name: 'Admin8' }).click();
+    await page.getByRole('menuitem', { name: 'Profile' }).click();
+    await page.getByRole('tab', { name: 'Security' }).click();
+    await page.getByPlaceholder('currentPassword').click();
+    await page.getByPlaceholder('currentPassword').fill('1234567Ak');
+    await page.getByPlaceholder('Password', { exact: true }).click();
+    await page.getByPlaceholder('Password', { exact: true }).fill('1234512345Ad');
+    await page.getByPlaceholder('passwordConfirmation').click();
+    await page.getByPlaceholder('passwordConfirmation').fill('1234512345Ad');
+    await page.getByRole('button', { name: 'Change Password' }).click();


There is no assertion (expect) in this test block.

________________________________

In packages/playwright/src/tests/profile-editing.spec.ts<https://github.com/freedomcombination/monorepo/pull/1579#discussion_r1717441471>:

> +    await page.getByRole('button', { name: 'EN' }).click();
+    await page.getByRole('link', { name: 'Sign in' }).click();
+    await page.waitForTimeout(1000)
+    await page.getByTestId('input-email').click();
+    await page.getByTestId('input-email').fill('Admin8');
+    await page.getByTestId('input-password').click();
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();
+    await page.evaluate(() => window.scrollTo(0, 0));
+    await page.getByRole('button', { name: 'Admin8' }).click();
+    await page.getByRole('menuitem', { name: 'Profile' }).click();
+    await page.getByRole('tab', { name: 'Socials' }).click();
+    await page.getByPlaceholder('linkedin').click();
+    await page.getByPlaceholder('linkedin').fill('https://www.linkedin.com/in/williamhgates/');
+    await page.evaluate(() => { window.scrollTo(0, document.body.scrollHeight); });
+    await page.getByRole('button', { name: 'Save' }).click();


There is no assertion (expect) in this test block.

________________________________

In packages/playwright/src/tests/profile-editing.spec.ts<https://github.com/freedomcombination/monorepo/pull/1579#discussion_r1717442149>:

> +    await page.goto(getVercelUrl('kunsthalte'))
+    await page.getByRole('button', { name: 'EN' }).click();
+    await page.getByRole('link', { name: 'Sign in' }).click();
+    await page.getByTestId('input-email').click();
+    await page.getByTestId('input-email').fill('Admin8');
+    await page.getByTestId('input-password').click();
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();
+    await page.evaluate(() => window.scrollTo(0, 0));
+    await page.getByRole('button', { name: 'Admin8' }).click();
+    await page.getByRole('menuitem', { name: 'Profile' }).click();
+    await page.getByRole('tab', { name: 'Socials' }).click();
+    await page.getByPlaceholder('linkedin').click();
+    await page.getByPlaceholder('linkedin').fill('https://www.linkedin.com/in/williamhgates');
+    await page.evaluate(() => { window.scrollTo(0, document.body.scrollHeight); });
+    await page.getByRole('button', { name: 'Save' }).click();


There is no assertion (expect) in this test block.

________________________________

In packages/playwright/src/tests/profile-editing.spec.ts<https://github.com/freedomcombination/monorepo/pull/1579#discussion_r1717442635>:

> +    await page.evaluate(() => { window.scrollTo(0, document.body.scrollHeight); });
+    await page.getByRole('button', { name: 'Save' }).click();
+    await page.evaluate(() => window.scrollTo(0, 0));
+    await page.getByRole('button', { name: 'Admin8' }).click();
+    await page.getByRole('menuitem', { name: 'Sign Out' }).click();
+  });
+
+  test ('User can display new credentials/ security ', async ({ page }) => {
+    await page.goto(getVercelUrl('kunsthalte'))
+    await page.getByRole('button', { name: 'EN' }).click();
+    await page.getByRole('link', { name: 'Sign in' }).click();
+    await page.getByTestId('input-email').click();
+    await page.getByTestId('input-email').fill('Admin8');
+    await page.getByTestId('input-password').click();
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();


There is no assertion (expect) in this test block.

________________________________

In packages/playwright/src/tests/profile-editing.spec.ts<https://github.com/freedomcombination/monorepo/pull/1579#discussion_r1717446649>:

> +    await page.getByRole('menuitem', { name: 'Sign Out' }).click();
+  });
+
+  test ('User can display new credentials/ socials ', async ({ page }) => {
+    await page.goto(getVercelUrl('kunsthalte'))
+    await page.getByRole('button', { name: 'EN' }).click();
+    await page.getByRole('link', { name: 'Sign in' }).click();
+    await page.getByTestId('input-email').click();
+    await page.getByTestId('input-email').fill('Admin8');
+    await page.getByTestId('input-password').click();
+    await page.getByTestId('input-password').fill('1234512345Ad');
+    await page.getByTestId('button-submit-login').click();
+    await page.getByRole('button', { name: 'Admin8' }).click();
+    await page.getByRole('menuitem', { name: 'Profile' }).click();
+    await page.getByRole('tab', { name: 'Socials' }).click();
+    const page1Promise = page.waitForEvent('popup');


What exactly is this promise and popup event for?

If it's necessary, I would give a more descriptive name to this variable, like: popupEventPromise, waitingPopupPromise etc.

File: packages/playwright/src/tests/profile-editing.spec.ts (hunk 3 above)
Comment (7alip) [reply packages/playwright/src/tests/profile-editing.spec.ts:53]: > Yes, the test was failing with this assertion but I found a new assertion instead.

File: packages/playwright/src/tests/profile-editing.spec.ts (hunk 1 above)
Comment (7alip) [reply packages/playwright/src/tests/profile-editing.spec.ts:26]: > The case that the user can log in with the updated password was requested in the second acceptance criteria under ‘after editing able to access new credentials’.
That's why I tested it under the ‘User can display new credentials/ security’ test header.

**Comment by 7alip** [comment]
@hilalnigarbasan I'll merge this into dev. I'm preparing dev environment for upcoming test cases.

**Pull request merged**